src/nvme/mi.c

```diff
@@ -414,6 +432,66 @@ static int nvme_mi_verify_resp_mic(struct nvme_mi_resp *resp)
 	return resp->mic != ~crc;
 }
 
+int nvme_mi_async_read(nvme_mi_ep_t ep, struct nvme_mi_resp *resp)
+{
+	int rc = ep->transport->aem_read(ep, resp);
+
+	if (nvme_mi_ep_has_quirk(ep, NVME_QUIRK_MIN_INTER_COMMAND_TIME))
+		nvme_mi_record_resp_time(ep);
+
+	if (rc && errno == EWOULDBLOCK) {
```

Comment:
`errno` may be overwritten by the `clock_gettime` call in `nvme_mi_record_resp_time` there. You may want to save it, or shift the record_resp_time call.

Given the read is non-blocking, you should be fine to record the time on function entry.